lib/enum.js

```diff
@@ -1,61 +1,54 @@
 'use strict';
 
-class Enum {}
+const NaE = Symbol('NotAnEnum');
 
-const enumArray = values => class extends Enum {
-  constructor(val) {
-    super();
-    this.value = this.constructor.key(val);
-  }
-  static get collection() {
-    return values;
-  }
-  static has(val) {
-    return values.includes(val);
-  }
-  static key(val) {
-    return values.includes(val) ? val : undefined;
-  }
-  [Symbol.toPrimitive]() {
-    return this.value;
-  }
-};
-
-const enumCollection = values => {
-  const index = {};
-  for (const key in values) {
-    const value = values[key];
-    index[value] = key;
-  }
-  return class extends Enum {
-    constructor(val) {
-      super();
-      this.value = this.constructor.key(val);
-    }
-    static get collection() {
-      return values;
-    }
-    static has(val) {
-      return !!(values[val] || index[val]);
+class Enum {
+  static from(...args) {
+    let values = args[0];
+    if (typeof values !== 'object' && !Array.isArray(values)) {
```

Comment:
The second check seems redundant since if the first check succeeds, then the value has a type other than `object` and how can it possibly be an array then?